intezer_sdk/analyses_results.py

```diff
@@ -0,0 +1,83 @@
+from typing import List
+from typing import Any
+from typing import Dict
+from typing import Tuple
+
+from intezer_sdk.api import IntezerApi
+from intezer_sdk.api import raise_for_status
+
+
+class AnalysesHistoryResult:
+    def __init__(self, request_url_path: str, api: IntezerApi, filters: Dict):
+        """
+        Fetch all analyses history results from server.
+        :param request_url_path: Url to request new filter from.
+        :param api: Instance of Intezer API for request server.
+        :param filters: Filters requested from server.
+        """
+        self.api: IntezerApi = api
+        self.filters: Dict = filters
+        self._pages: List[Any] = []
+        self._current_page: List[Any] = []
+        self._request_url_path: str = request_url_path
+        self._current_page_number: int = 0
+        self._total_count: int = 0
+        self._current_offset: int = 0
+
+    def __iter__(self):
+        """Iterate between page."""
+        if not self._current_page:
+            self._fetch_page()
+        if self._current_page:
+            yield from self._current_page
+            if len(self._pages) * self.filters['limit'] < self._total_count:
+                self._fetch_page()
+                yield from iter(self)
+
+    def __len__(self) -> int:
+        """Amount of results fetched currently."""
+        return self._total_count
+
+    def current_page(self) -> List:
+        """Get current page, if not exits, ask a new one from server."""
+        return self._current_page or self._fetch_page()
+
+    def all(self) -> List:
+        """List all remaining and exists analysis's from server."""
```

Comment:
Is it the existing or the remaining?

Reply:
All will give you everything, it will fetch the remaining and will append to the exists analysis's.